### PR TITLE
Streaming API + Tests

### PR DESCRIPTION
**Pull request by iostat**
Tracking PR, please ignore. 

Merges #2 and tests it.

### CODE REVIEW

**Comment by martyall** [comment]
just as a first note, i noticed that jq and presumably truffle are dependencies here. should we update the README to reflect the tests?

**Comment by martyall** [comment]
screen shot 2017-12-27 at 3 12 04 pm

@iostat  I can't remember which ones of these we said are supposed to fail, and which ones aren't if they are supposed to fail, maybe we should mark them as `pending` for now?

**Comment by iostat** [comment]
The two SimpleStorage tests are supposed to fail. And yeah, good point re:
truffle and jq.

Also, I’ve noticed that I’m getting unexpected results (events that make
absolutely no sense, eg CountSet 0 and 48 when i’ve set 1, 2, and 3) with
the new streaming API. About to push those cases.

File: Setup.hs [hunk 1]
```diff
@@ -36,8 +36,4 @@ myBuildHook pd lbi uh flags = do
     buildHook simpleUserHooks pd lbi uh flags
 
     where v = fromFlag $ buildVerbosity flags
-
-myTestHook :: Args -> PackageDescription -> LocalBuildInfo -> UserHooks -> TestFlags -> IO ()
-myTestHook args pd lbi uh tf = do
-    putStrLn "my test hook!" -- this apparently doesnt run?
-    -- testHook simpleUserHooks args pd lbi uh tf
+          hasTestTarget = any ("test:" `isPrefixOf`) $ buildArgs flags
```
Comment (iostat) [inline Setup.hs:38]: if it's stupid but it works, it's not stupid!

**Pull request closed without merging**